hue/hue.sh

```diff
@@ -24,127 +29,142 @@ function update_apt_get() {
   return 1
 }
 
-# Determine the role of this node
-ROLE=$(/usr/share/google/get_metadata_value attributes/dataproc-role)
-
-# Only run on the master node of the cluster
-if [[ "${ROLE}" != 'Master' ]]; then
-  exit 0
-fi
-
+function install_hue_and_configure(){
+local old_hdfs_url='## webhdfs_url\=http:\/\/localhost:50070'
+local new_hdfs_url='webhdfs_url\=http:\/\/'"$(hdfs getconf -confKey  dfs.namenode.http-address)"
+local old_mysql_settings='## engine=sqlite3(\s+)## host=(\s+)## port=(\s+)## user=(\s+)## password='
+local new_mysql_settings='engine=mysql$1host=127.0.0.1$2port=3306$3user=hue$4password=hue-password'
+local mysql_password='root-password'
+local hue_password='hue-password'
 # Install hue
-update_apt_get
-apt-get install -t jessie-backports hue -y
+apt-get install -t jessie-backports hue -y || err "Failed to install hue"
 
 # Stop hue
-systemctl stop hue
-
-cat > core-site-patch.xml <<EOF
-<property>
-  <name>hadoop.proxyuser.hue.hosts</name>
-  <value>*</value>
-</property>
-<property>
-  <name>hadoop.proxyuser.hue.groups</name>
-  <value>*</value>
-</property>
-EOF
-sed -i '/<\/configuration>/e cat core-site-patch.xml' \
-   /etc/hadoop/conf/core-site.xml
+systemctl stop hue || err "Hue stop action not performed"
 
+bdconfig set_property \
+  --configuration_file 'core-site-patch.xml' \
```

Comment:
You can just avoid using `core-site-patch.xml` throughout this section and instead just directly pass `--configuration-file=/etc/hadoop/conf/core-site.xml`. That gets rid of the confusing sed below.

Reply:
@karth295 Could you explain this in a more detailed way?